src/averageAges.js

```diff
@@ -55,12 +70,34 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const peopleWeNeed = onlyWithSon
+    ? people.filter(person =>
+      person.sex === 'm' && people.some(child => child.name === person.mother))
+    : people.filter(person =>
+      people.some(child => child.name === person.mother));
+
+  const sumOfAgeDiffs = peopleWeNeed.reduce((sum, child) => {
+    const mother = people.find(person => person.name === child.mother);
+
+    if (mother) {
+      const ageDiff = child.born - mother.born;
+
+      return sum + ageDiff;
+    }
+
+    return sum;
+  }, 0);
+
+  const averageAgeDiff = sumOfAgeDiffs / peopleWeNeed.length;
+
+  return averageAgeDiff;
 }
 
+// 1. find a mother of each person (or only for men)
```

Comment:
The comments appear to be instructions for the function above them. They should be moved above the function `calculateAverageAgeDiff` they're describing.